test/jasmine/tests/shapes_test.js

```diff
@@ -1504,18 +1510,18 @@ describe('Test shapes', function() {
 
     function getShapeCoordinates(layoutShape, x2p, y2p) {
         return {
-            x0: x2p(layoutShape.x0),
-            x1: x2p(layoutShape.x1),
-            y0: y2p(layoutShape.y0),
-            y1: y2p(layoutShape.y1)
+            x0: x2p(layoutShape.x0, layoutShape.x0shift),
+            x1: x2p(layoutShape.x1, layoutShape.x1shift),
+            y0: y2p(layoutShape.y0, layoutShape.y0shift),
+            y1: y2p(layoutShape.y1, layoutShape.y1shift)
         };
     }
 
     function testPathDrag(dx, dy, layoutShape, node) {
         var xa = Axes.getFromId(gd, layoutShape.xref);
         var ya = Axes.getFromId(gd, layoutShape.yref);
-        var x2p = helpers.getDataToPixel(gd, xa);
-        var y2p = helpers.getDataToPixel(gd, ya, true);
+        var x2p = helpers.getDataToPixel(gd, xa, layoutShape.xshift);
+        var y2p = helpers.getDataToPixel(gd, ya, layoutShape.yshift, true);
```

Comment:
Do we have `xshift` and `yshift` on the `layoutShape` here?

Reply:
actually no, the function that creates the shapes does not set a shift. Should I just add xshift and yshift 0 to the ` setupLayout` function?

Reply:
If required yes.

Reply:
Yeah, so right now they just evaluate to "undefined" because they are not set, which the function handles fine. But yeah, might be clearer.

Reply:
Nvm, now I get your question and updated the code. I just overlooked the remaining places where xshift and yshift were still used instead of x0/x1/y0/y1shift.